### PR TITLE
Split SonarCloud in two workflows

### PR DESCRIPTION
**Pull request by fmarco76**
The pull request fork run in a separate workflow triggered by the main
solarcloud workflow.

NOTE: additional tests for pull requests to secondary branch should be
done but these are less frequent

### CODE REVIEW

**Comment by fmarco76** [comment]
In order to work properly, the pull workflow has to be in the master branch because it can be run only from the master so we can test only after this is merged.

**Comment by fmarco76** [comment]
For reference, this is based on: https://securitylab.github.com/research/github-actions-preventing-pwn-requests/

**Review by ckelleyRH — APPROVED** [review]
LGTM. Let's merge and see how it goes!

File: .github/workflows/sonarcloud.yml [hunk 1]
```diff
@@ -45,3 +46,36 @@ jobs:
...
+        uses: actions/cache@v3
+        with:
+          key: pki-sonar-runner-${{ matrix.os }}-${{ github.run_id }}
+          path: pki-sonar-runner.tar
+
+      - name: Upload build as artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: pki-build
+          path: pki-sonar-runner.tar
```
Comment (edewata) [inline .github/workflows/sonarcloud.yml:81]: Do we need to keep the container image? Usually we just keep the logs, but it's up to you.

**Review by edewata — APPROVED** [review]
This is too advanced for me :)
I just have one question, but feel free to merge.

File: .github/workflows/sonarcloud.yml (hunk 1 above)
Comment (fmarco76) [reply .github/workflows/sonarcloud.yml]: I have removed the upload. The build is done in the second workflow so the only thing uploaded is the PR number, which is only few bytes.

I am merging and let see if it work like in my repo.

**Pull request merged**